client/src/components/MobileVirtualJoystick.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useLayoutEffect, useState } from 'react'
+import styled from 'styled-components'
+import JoystickItem from './Joystick'
+
+import phaserGame from '../PhaserGame'
+import Game from '../scenes/Game'
+
+import { useAppSelector } from '../hooks'
+import { JoystickMovement } from './Joystick'
+
+const Backdrop = styled.div`
+  position: fixed;
+  bottom: 100px;
+  right: 32px;
+  max-height: 50%;
+  max-width: 100%;
+`
+
+const Wrapper = styled.div`
+  position: relative;
+  height: 100%;
+  padding: 16px;
+  display: flex;
+  flex-direction: column;
+`
+
+const JoystickWrapper = styled.div`
+  margin-top: auto;
+  align-self: flex-end;
+`
+export const smallScreenSize = 650 // minimum width for small screen
+
+function useWindowSize() {
+  const [size, setSize] = useState([0, 0])
+  useLayoutEffect(() => {
+    function updateSize() {
+      setSize([window.innerWidth, window.innerHeight])
+    }
+    window.addEventListener('resize', updateSize)
+    updateSize()
+    return () => window.removeEventListener('resize', updateSize)
+  }, [])
+  return size
+}
+
+const isSmallScreen = () => {
+  const [width] = useWindowSize()
+  return width <= smallScreenSize
+}
+
+export default function MobileVirtualJoystick() {
+  const showJoystick = useAppSelector((state) => state.user.showJoystick)
+  const showChat = useAppSelector((state) => state.chat.showChat)
+  const game = phaserGame.scene.keys.game as Game
+
+  useEffect(() => {}, [showJoystick, showChat])
+
+  const handleMovement = (movement: JoystickMovement) => {
+    game.myPlayer?.handleJoystickMovement(movement)
+  }
+
+  return (
+    <Backdrop>
+      <Wrapper>
+        {!(showChat && isSmallScreen) && showJoystick && (
```

Comment:
`isSmallScreen` is a function but it doesn't seem like you're calling it, some typo here?

Reply:
Uff you'r right 😅, while I'm at it, I'll take the opportunity to make this cleaner